docs/topics/development/setup_and_configuration.md

```diff
@@ -0,0 +1,316 @@
+# Setup and Configuration
+
+## Local Development Environment
+
+Setting up the local development environment for the **addons-server** involves configuring Docker Compose to run the necessary services. Follow these steps to get started:
+
+1. **Prerequisites**:
+    - Ensure Docker and Docker Compose are installed on your system.
+    - Clone the **addons-server** repository from GitHub:
+
+      ```sh
+      git clone https://github.com/mozilla/addons-server
+      cd addons-server
+      ```
+
+2. **Configuration**:
+    - Run the following command to create the `.env` file with necessary environment variables:
+
+      ```sh
+      make setup
+      ```
+
+      This command generates the `.env` file required by our Docker Compose configuration in the `env_file` property. The `.env` file includes sensible defaults for each required value. Changes to the `.env` file will persist if you rerun `make setup`.
+
+3. **Bringing Up Services**:
+    - When running the project for the first time, execute:
+
+      ```sh
+      make initialize_docker
+      ```
+
+      This command will:
+      - Run `make up` to start the Docker containers.
+      - Run `make initialize` to set up the initial Docker environment, including database initialization and data population. Detailed steps for `make initialize` will be covered in Section 6 on Data Management.
+
+4. **Understanding `make up`**:
+    - The `make up` command ensures all necessary files are created on the host and starts the Docker Compose project, including volumes, containers, and networks.
+    - Here’s a high-level overview of what `make up` does:
+
+      ```sh
+      up: setup docker_mysqld_volume_create docker_extract_deps docker_compose_up
+      ```
+
+      - **setup**: Creates configuration files such as `.env`.
+      - **docker_mysqld_volume_create**: Ensures the MySQL volume is created.
+      - **docker_extract_deps**: Installs dependencies inside the Docker container.
+      - **docker_compose_up**: Starts the Docker containers defined in [docker-compose.yml][docker-compose].
+
+5. **Accessing the Development Environment**:
+    - Add the following entry to your `/etc/hosts` file to access **addons-server** via a local domain:
+
+      ```sh
+      127.0.0.1 olympia.test
+      ```
+
+    - The web application should now be accessible at `http://olympia.test`.
+    - You can access the web container for debugging and development:
+
+      ```sh
+      make shell
+      ```
+
+    - To access the Django shell within the container:
+
+      ```sh
+      make djshell
+      ```
+
+### Configuring your environment
+
+#### Build vs Pull
```

Comment:
I didn't really follow this section - it's named build vs pull but it wasn't clear to me what exact steps you'd take to get a "build" vs. a "pull" scenario.  Or is pull in the next section (that is named Version vs Digest) instead?  😕

Reply:
wdyt?